utils/preferences/tests/dtf/preferences.rs

```diff
@@ -0,0 +1,35 @@
+use icu_locid::{extensions::unicode, extensions_unicode_value};
+use tinystr::{tinystr, TinyStr4};
+
+#[derive(Clone, Copy)]
+pub enum Calendar {
+    Gregory,
+    Buddhist,
+}
+
+impl TryFrom<&unicode::Value> for Calendar {
+    type Error = ();
+
+    fn try_from(i: &unicode::Value) -> Result<Self, Self::Error> {
+        match i {
+            _ if *i == extensions_unicode_value!("gregory") => Ok(Self::Gregory),
+            _ if *i == extensions_unicode_value!("buddhist") => Ok(Self::Buddhist),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+pub struct NumberingSystem(pub TinyStr4);
+
+impl TryFrom<&unicode::Value> for NumberingSystem {
+    type Error = ();
+
+    fn try_from(i: &unicode::Value) -> Result<Self, Self::Error> {
+        match i {
+            _ if *i == extensions_unicode_value!("latn") => Ok(Self(tinystr!(4, "latn"))),
+            _ if *i == extensions_unicode_value!("arab") => Ok(Self(tinystr!(4, "arab"))),
```

Comment:
Nit: I would rather not check for exact values here. I'd rather stick with the space of potentially valid values specified by Unicode. I would need to check whether this is a more limited set than "a valid subtag string".